src/backend-project/backend-project.service.ts

```diff
@@ -67,8 +69,16 @@ export class BackendProjectService {
           // add remote and fetch repo
           return this.addRemoteAndFetchRepo(localRepo, remoteRepo, remoteName, remoteBranch);
         }
-        debug(`yep - we have remote ${remoteName}; fetching branch ${remoteBranch} of ${localRepo} from ${remoteName}`);
-        return this.gitService.fetch(localRepo, remoteName, remoteBranch).pipe(
+        debug(`yep - we have remote ${remoteName}; fetching ${localRepo} and pulling branch ${localBranch} from ${remoteName}`);
+        return this.gitService.fetch(localRepo, remoteName).pipe(
+          switchMap(() => this.gitService.pull(localRepo, remoteName, localBranch)),
+          catchError(err => {
+            // this can happen when running e2e tests
+            if (err.message.indexOf(`couldn't find remote ref ${localBranch}`) >= 0) {
```

Comment:
Can you explain why this would happen during e2e tests?

Also, this is fairly fragile as it is parsing a git error message that could change at any time, right?

Reply:
During the e2e tests we run several similar tests. Between the tests we reset the repos, both locally and on GitHub. To safe time and in order not having to download >1GB of data for **each** test we don't delete the local clone between the tests.

Yes, it is a bit fragile, but the worst that can happen is that a e2e test fails.